src/components/share-overlay/share-overlay.js

```diff
@@ -707,7 +716,14 @@ class ShareOverlay extends Component {
       <div className={this.state.view === shareOverlayView.EmbedOptions ? 'overlay-screen active' : 'overlay-screen'}>
         <div className={style.title}>{props.title}</div>
         <div className={shareStyle.linkOptionsContainer}>
-          <ShareUrl addAccessibleChild={this.props.addAccessibleChild} shareUrl={props.shareUrl} copy={true} isIos={this.isIos} />
+          <ShareUrl
+            player={this.props.player}
```

Comment:
Passing big instances directly via props is ill advised, passing the player is not recommend

Reply:
@OrenMe Theoretically, I agree, 
But it is already implemented (here and in almost all the rest of old plugins ) in this way in the other components as well, but nothing in this plugin is recommended... and is not badly written, ideally every plugin would be a FakeEventTarget by itself (either by composition or inheritance) and (in a decoupled way) propagate its events and not use the instance instance to fire, but as you can see how this plugin looks ..

 in any case what’s your alternative suggestion? Passing it as event handler from the plugin class all the way down to this comp?

Reply:
we already have HoC to add even system afaicr
if you only need this for events, pass onEvent to your component and only use the main plugin entry file to publish them outside
The main plugin already has event target
When u pass the player like this the props comparison is huge and u get too many re-renders

Reply:
We have event listeners manager Hoc, not Event dispatcher

Reply:
See usage of other ui code in playkit UI, we exposed events long ago but in any case the other solution is better to expose the events only via the plugin which is already an event target and also it saves u on invalid passing of player as prop

Reply:
After a thorough review,  since it is pointless and inconsistent  to correct in only one place (as you can see the player is already passed down the components in the other components, this will be fixed in a separate PR as part of the longed planned refactor here